packages/visualizations/src/components/Table/Table.svelte

```diff
@@ -12,12 +13,44 @@
     export let emptyStateLabel: string | undefined;
 
     const tableId = `table-${generateId()}`;
+
+    let isHorizontallyScrolled = false;
+    let scrollBox: HTMLDivElement;
+    let sortedStickyColumns: Column[] = [];
+
+    function handleScroll() {
+        isHorizontallyScrolled =
+            document.dir === 'rtl' ? scrollBox?.scrollLeft < 0 : scrollBox?.scrollLeft > 0;
+    }
+
+    // resets scroll when changing columns parameters
+    $: if (columns && scrollBox) {
+        scrollBox.scrollLeft = 0;
+        sortedStickyColumns = [...columns].sort((colA, colB) => {
+            if (Boolean(colA?.sticky) === Boolean(colB?.sticky)) {
+                return 0;
+            }
+            return colA?.sticky ? -1 : 1;
+        });
+        stickyColumnsWidth.reset();
+        sortedStickyColumns
+            .filter((col) => col?.sticky)
+            .forEach((col) => {
+                stickyColumnsWidth.updateColumn(col.key, 0);
+            });
+    }
```

Comment:
```suggestion
    $: if (columns && scrollBox) {
        scrollBox.scrollLeft = 0;
        stickyColumnsWidth.reset();
        columns.forEach((col) => {
                col?.sticky && stickyColumnsWidth.updateColumn(col.key, 0);
            });
    }
```

Did I miss something?
Do we want to get update the store state with only columns keys that are sticky?
(While keeping the original order of the column)

Reply:
I opted to sort the sticky column first so that they are already in their sticky position. In the version you shared colmun will stick "as they come". I thought it was pretty common for "pinned" columns to be first. This could also be done by putting sticky columns in a different array (in which case we don't need to sort).

Reply:
```js
[...columns].sort((colA, colB) => {
            if (Boolean(colA?.sticky) === Boolean(colB?.sticky)) {
                return 0;
            }
            return colA?.sticky ? -1 : 1;
        });
```

```js
columns.filter((col) => col?.sticky)
```

seems to produce the same result.
I don't understand why we need this sort function